### PR TITLE
refactor(deps): Remove dependency on toml_edit (closes #78)

### LINKED ISSUE #78: Replace toml_edit
`toml_edit` is another fairly heavy dependency that I pulled in early to make dev easier. We can probably achieve the same thing with the normal `toml` lib which we also depend on, just gotta make sure we preserve key order so we don't break the same thing that Zak fixed in #75 .

### PR DESCRIPTION
**Pull request by dbanty**
Based on #77 so that needs to get merged first (avoiding cargo conflicts). We miss out on the benefits in toml_edit of retaining white space and formatting, so maybe this isn't worth it? But toml_edit is one of our biggest dependencies right now.

### Stats

| Stat | Before | After |
| ---- | ------ | ------ |
| # Deps | 186 | 170 |
| Release Build Time | 2m 11s | 1m 45s |
| Release Bin Size | 6.4 MB | 5.9 MB |

### CODE REVIEW

**Review by Shadow53 — CHANGES_REQUESTED** [review]
So, in context, we have -26 seconds in release build time and -500KB in executable size, in exchange for what I think is messier code, or at least less concise. Keep in mind that 26 seconds should not affect development time (it should run on GH Actions when a release is tagged) and 500KB is not much: about half an average MP3 or the size of a poor-quality JPEG.

As a reminder, the use of `toml_edit` to maintain whitespace and formatting is primarily to prevent unnecessarily large diffs for something like a version bump. So if we weigh that benefit against the measured gains from dropping the dependency... I personally think the benefit wins out and this PR should be dropped.

I am not the maintainer, though, so I ultimately defer to @dbanty for the decision. If you decide to push this PR through, I would like to request that a more generic helper function be made to make the `set_version` functions more concise. Something like:

```
let mut toml: toml::Value = toml::from_str(&std::fs::read_to_string(&path)?)?;
set_toml_key(&mut toml, ["tool", "poetry", "version"], toml::Value::String(new_version))?;
```

This would standardize the error messages and deduplicate a bit of code.

**Comment by dbanty** [comment]
I think you're right in that preserving whitespace is a feature worth... preserving. 26s does make a significant difference since it's a cost most users will have to pay on `cargo install` right now, but the better strategy to pursue, I think, is improving the install process by making this available in binary form on package managers (e.g. homebrew).

**Pull request closed without merging**
